taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -526,11 +542,19 @@ func (self *Contract) DistributeRewards(rewardsStats *rewards_stats.RewardsStats
 		}
 
 		// Calculate it like this to eliminate rounding error as much as possible
-		validatorReward := bigutil.Mul(big.NewInt(int64(validatorStats.UniqueTxsCount)), blockReward)
+		validatorReward := bigutil.Mul(big.NewInt(int64(validatorStats.UniqueTxsCount)), trxReward)
 		validatorReward = bigutil.Div(validatorReward, big.NewInt(int64(rewardsStats.TotalUniqueTxsCount)))
 
-		// TODO: once we have also votes statistics, use it in calculations
-		// TODO: should voter with 1M stake be rewarded same as voter with 10M stake ???
+		// Add reward for voting
+		if validatorStats.ValidCertVote {
+			totalUniqueVoteCountCheck++
+			validatorReward = bigutil.Add(validatorReward, voteReward)
+		}
+
+		// Add reward to the author for additional included votes
+		if validatorAddress == *author {
```

Comment:
probablly bug ? What if pbft block author did not create any dag block nor he voted ? It should be handled separately outside of this for loop